File: php/difference-of-squares/difference-of-squares.php
<?php

function squareOfSum($n) {
  return array_sum(range(1, $n)) ** 2;
}

function sumOfSquares($n) {
  return array_reduce(range(1, $n), function($acc, $curr) {
    return $acc + $curr ** 2;
  }, 0);
}

function difference($n) {
  return squareOfSum($n) - sumOfSquares($n);
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
import kotlin.math.pow

class Squares(private val limit: Int) {

    fun sumOfSquares() = (1..limit).sumBy { it * it }

    fun squareOfSum() = (1..limit).sum().toDouble().pow(2).toInt()

    fun difference() = squareOfSum() - sumOfSquares()
}
